solr/core/src/test/org/apache/solr/cloud/TestConfigSetsAPI.java

```diff
@@ -936,7 +936,7 @@ private void testNewSingleFileAfterSchemaAPI(boolean v2) throws Exception {
     String collectionName = "newcollection";
     String configsetName = "regular";
     String configsetSuffix = "testSinglePathNew-1-" + v2;
-    createConfigSet(null, configsetName + configsetSuffix, null, cluster.getSolrClient(), "solr");
+    uploadConfigSetWithAssertions(configsetName, configsetSuffix, "solr");
```

Comment:
isn't createConfigSet synchronous?  What does this PR perturb to cause this?

Reply:
Good point @dsmiley  - my guess was wrong here.

After looking into the test failure more, I realized the issue happens because the test adds a field to the schema. The schema modification triggers the ZkIndexSchemaReader$SchemaWatcher watch - but while the watch is running, the test may already finish and the test suite may start deleting configsets from ZK.

I reverted the change to the test, and instead modified the SchemaWatcher code to gracefully handle such scenario.